pkg/apis/serving/v1alpha1/revision_types_test.go

```diff
@@ -576,3 +579,150 @@ func TestRevisionGetGroupVersionKind(t *testing.T) {
 		t.Errorf("got: %v, want: %v", got, want)
 	}
 }
+
+func TestRevisionGetLastPinned(t *testing.T) {
+	cases := []struct {
+		name        string
+		annotations map[string]string
+		expectTime  time.Time
+		expectErr   error
+	}{{
+		name:        "Nil annotations",
+		annotations: nil,
+		expectTime:  time.Time{},
+		expectErr: LastPinnedParseError{
+			Type: AnnotationParseErrorTypeMissing,
+		},
+	}, {
+		name:        "Empty map annotations",
+		annotations: map[string]string{},
+		expectTime:  time.Time{},
+		expectErr: LastPinnedParseError{
+			Type: AnnotationParseErrorTypeMissing,
+		},
+	}, {
+		name:        "Invalid time",
+		annotations: map[string]string{serving.RevisionLastPinnedAnnotationKey: "abcd"},
+		expectTime:  time.Time{},
+		expectErr: LastPinnedParseError{
+			Type:  AnnotationParseErrorTypeInvalid,
+			Value: "abcd",
+		},
+	}, {
+		name:        "Valid time",
+		annotations: map[string]string{serving.RevisionLastPinnedAnnotationKey: "10000"},
+		expectTime:  time.Unix(10000, 0),
+		expectErr:   nil,
+	}}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			rev := Revision{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: tc.annotations,
+				},
+			}
+
+			pt, err := rev.GetLastPinned()
+			expectErr, _ := tc.expectErr.(LastPinnedParseError)
```

Comment:
Just type `tc.expectErr` as `LastPinnedParseError`?  Should this be `(*LastPinnedParseError)`?

Reply:
Reworked to avoid the casting altogether.